### ISSUE TITLE
Can't timestamp my self-signed authenticode certificate

### ISSUE THREAD

**Issue by CaptainFrosty** [PowerShell/PowerShell#19577]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Attempt ` Set-AuthenticodeSignature -FilePath D:\Documents\PowerShell\Microsoft.Powershell_profile.ps1 -Certificate $cert -HashAlgorithm "SHA256" -TimestampServer "http://timestamp.verisign.com/scripts/timstamp.dll"`

The file shows no timestamp on the signature no matter which timestamp server I try.

### Expected behavior

```console
A Timestamp
```

### Actual behavior

```console
A successful signing with no timestamp.
```

### Error details

```console
N/A
```

### Environment data

```powershell
PSVersion                      7.4.0-preview.3
PSEdition                      Core
GitCommitId                    7.4.0-preview.3
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

I also tried on the system PowerShell.
```

**Comment by PaulHigin** [comment]
For some reason the code is masking the `ERROR_INTERNET_NAME_NOT_RESOLVED` error returned if the timestamp server url could not be resolved.  So the server may be invalid and the user would never know.  Apparently this was to prevent a breaking change, but it seems like the user should at least be informed via a warning message.

```csharp
// CryptUIWizDigitalSign introduced a breaking change in Win8 to return this
// error code (ERROR_INTERNET_NAME_NOT_RESOLVED) when you provide an invalid
// timestamp server. It used to be 0x80070001.
// Also masking this out so that we don't introduce a breaking change ourselves.
```
https://github.com/PowerShell/PowerShell/blob/ea68edc686d5f2370041aa9284805d976d9c7553/src/System.Management.Automation/security/Authenticode.cs#L222

**Comment by CaptainFrosty** [comment]
Thanks for the response. I figured the lack of error was something like that; it does silently complete as if everything worked, and I doubt it is the syntax as that usually gets caught right away. 

I was stumped after trying several servers, but I may end up feeling dumb. I just remembered about my ESET Internet Security that I keep on a "policy mode" which means it force filters all SSL/TLS connections, and some connections do not like this friendly man in the middle. Hopefully it will just be a matter of setting ESET to exclude the certificate for the timestamp server URL to resolve it. As I will go find out now.
